src/gt4py/gtscript.py

```diff
@@ -171,6 +166,8 @@ def stencil(
         raise ValueError(f"Invalid 'name' string ('{name}')")
     if not isinstance(rebuild, bool):
         raise ValueError(f"Invalid 'rebuild' bool value ('{rebuild}')")
+    if _dev_opts is not None and not isinstance(_dev_opts, dict):
+        raise ValueError(f"Invalid '_dev_opts' dictionary ('{_dev_opts}')")
```

Comment:
If we're only checking that it's a dict, the error could just say `'_dev_opts' must be a dict`

Reply:
Ok, I have changed it to a more complete check